app/controllers/staff/activity_forms_controller.rb

```diff
@@ -74,6 +77,9 @@ def update
     when :identifier
       @activity.assign_attributes(delivery_partner_identifier: delivery_partner_identifier)
       add_transparency_identifier
+    when :roda_identifier
+      @activity.assign_attributes(roda_identifier_fragment: roda_identifier_fragment)
+      @activity.cache_roda_identifier!
```

Comment:
Did you consider tying setting the compound column to the `roda_identifier_fragment=` setter method (and maybe making the compound attribute read-only)?

There's quite a few behaviours that only exist within the multi-step form that has to be duplicated into ingest, CSV import etc, and they're easy to miss.

Reply:
I did consider putting this logic in the model -- tying an update of `roda_identifier_compound` whenever `roda_identifier_fragment` is changed. However, I thought implementing it that way would remove a lot of context that's important for this change. If we put this logic in the model, it needs to respond to any situation when `roda_identifier_fragment` is changed, e.g. updating the `roda_identifier_compound` of all the child/descdendent activities.

I'd rather move this logic up a layer and embed it inside the workflow that checks whether you're allowed to set the RODA ID, which restricts the kinds of updates we're expecting and means `roda_identifier_compound` only needs to be updated on the current model.

I was worried that a more general-purspoe implementation could cause unexpected behaviour if someone actually needed to go in and change the `roda_identifier_fragment` via the console. Does that make sense?

Making `roda_identifier_compound` read-only is appealing -- is there a way of doing this so it can be set if it's `nil`, but not after that? Activities are not usually new records when this attribute is set.

Reply:
I've now added `Activity#roda_identifier_compound=` to catch and code that's mistakenly setting this field.